libs/pubsub/query/query.go

```diff
@@ -195,6 +197,28 @@ func (q *Query) Matches(events map[string][]string) bool {
 			op = OpEqual
 		case rulecontains:
 			op = OpContains
+		case ruleexists:
+			op = OpExists
+			if strings.Contains(tag, ".") {
+				// Searching for a full "type.attribute" event.
+				_, ok := events[tag]
+				if !ok {
+					return false
+				}
+			} else {
+				foundEvent := false
```

Comment:
```
for eventType := range events {
  if eventType == tag {
    return true
  }
}
return false
```

Reply:
It can't actually be simplified that much. It's important that on a positive match, the evaluation continues. Consider the query `slash.reason EXISTS AND slash.power > 1000` on an event where `slash.power` is `500`. First part of the query matches, but the second won't.

Reply:
Right, thanks!